detectable/src/main/java/com/synopsys/integration/detectable/detectables/conan/graph/ConanNodeBuilder.java

```diff
@@ -0,0 +1,193 @@
+/**
+ * detectable
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.detectable.detectables.conan.graph;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.StringTokenizer;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO when ConanNode is re-worked, this class will need to be completely  re-thought
+public class ConanNodeBuilder {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private String ref;
+    private String path;
+    private String name;
+    private String version;
+    private String user;
+    private String channel;
+    private String recipeRevision;
+    private String packageId;
+    private String packageRevision;
+    private final List<String> requiresRefs = new ArrayList<>();
+    private List<Integer> requiresIndices;
+    private final List<String> buildRequiresRefs = new ArrayList<>();
+    private List<Integer> buildRequiresIndices;
+    private boolean valid = true;
+    private boolean forcedRootNode = false;
+
+    public ConanNodeBuilder forceRootNode() {
+        forcedRootNode = true;
+        return this;
+    }
+
+    public ConanNodeBuilder setRefFromLockfile(String ref) {
+        if (StringUtils.isBlank(ref)) {
+            return this;
+        }
+        ref = ref.trim();
+        StringTokenizer tokenizer = new StringTokenizer(ref, "@/#");
```

Comment:
It seemed like parsing and model were well separated but here the builder does the parsing? I think ideally this would be in some kind of parser.

Reply:
good point. all parsing is now in the parsers